src/main/java/de/luhmer/owncloud/accountimporter/helper/AccountImporter.java

```diff
@@ -2,153 +2,147 @@
+import android.support.annotation.RequiresApi;
 
-import java.net.URL;
+import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
 
-import de.luhmer.owncloud.accountimporter.ImportAccountsDialogFragment;
-import de.luhmer.owncloud.accountimporter.interfaces.IAccountImport;
+import de.luhmer.owncloud.accountimporter.interfaces.IAccountsReceived;
+
+import static android.content.Context.MODE_PRIVATE;
 
 /**
  * Created by david on 28.05.14.
  */
 public class AccountImporter {
 
-    public static List<Account> findAccounts(Activity activity) {
-        final AccountManager accMgr = AccountManager.get(activity);
+    private static final String TAG = AccountImporter.class.getCanonicalName();
+    private static final String PREF_FILE_NAME = "PrefNextcloudAccount";
+    private static final String PREF_ACCOUNT_STRING = "PREF_ACCOUNT_STRING";
+
+    public static boolean AccountsToImportAvailable(Context context) {
+        return FindAccounts(context).size() > 0;
+    }
+
+    //TODO add multi account support
+    public static Account GetCurrentAccount(Context context) {
+        SharedPreferences preferences = context.getSharedPreferences(PREF_FILE_NAME, MODE_PRIVATE);
+        String accountName = preferences.getString(PREF_ACCOUNT_STRING, "");
+        return GetAccountForName(context, accountName);
+    }
+
+    //TODO add multi account support
+    public static SingleSignOnAccount GetCurrentSingleAccount(Context context) throws AuthenticatorException, OperationCanceledException, IOException {
+        return GetAuthToken(context, GetCurrentAccount(context));
+    }
+
+    //TODO add multi account support
+    public static void SetCurrentAccount(Context context, Account account) {
+        SharedPreferences preferences = context.getSharedPreferences(PREF_FILE_NAME, MODE_PRIVATE);
+        preferences.edit().putString(PREF_ACCOUNT_STRING, account.name).commit();
+    }
+
+
+    // Find all currently installed nextcloud accounts on the phone
+    private static List<Account> FindAccounts(Context context) {
+        final AccountManager accMgr = AccountManager.get(context);
         final Account[] accounts = accMgr.getAccounts();
 
-        List<Account> accountsAvailable = new ArrayList<Account>();
+        List<Account> accountsAvailable = new ArrayList<>();
         for (Account account : accounts) {
-            String aType = account.type.intern();
-
-            if (//aType.equals("org.dmfs.caldav.account") ||
-                // aType.equals("org.dmfs.carddav.account") ||
-                    aType.equals("de.luhmer.tasksync") ||
-                            aType.equals("owncloud")) {
-                //accountsAvailable.add(accounts[index].type);
+            if (account.type.equals("nextcloud")) {
```

Comment:
This  "nextcloud" thing needs to be configurable.

Reply:
Why? Aren't all nextcloud accounts stored with the type "nextcloud"?

Reply:
The files app is brandable, so e.g. if you want to change everything from Nextcloud to Ubercloud, also the account type is "ubercloud" as otherwise you would see the other accounts in your "new" app.

Reply:
Oh okay, I see. Any good ideas on how to change that accordingly? Another question would be.. how do we know which "brand" the user wants to use? 

Apart from that, I don't think that we need to check the type in the future since starting with Android 8,  the `accMgr.getAccounts();` returns only accounts that the app has access to. (https://developer.android.com/about/versions/oreo/android-8.0-changes#o-sec)

The client is basically the only app that needs to know about the account type then.. as seen here: https://github.com/nextcloud/news-android/blob/sso/News-Android-App/src/main/java/de/luhmer/owncloudnewsreader/LoginDialogFragment.java#L249

Reply:
For now this is not a big deal. Anything branded will just be files+talk I guess.